### ISSUE TITLE
AttributeError: 'module' object has no attribute 'e'

### ISSUE THREAD

**Issue by kmatzen** [ansible/ansible#9833]
##### Issue Type:

Bug Report
##### Ansible Version:

```
ansible 1.9 (devel fa6e587654) last updated 2014/12/17 00:10:50 (GMT +000)
  lib/ansible/modules/core: (detached HEAD 9c6826e928) last updated 2014/12/17 00:10:51 (GMT +000)
  lib/ansible/modules/extras: (detached HEAD 317654dba5) last updated 2014/12/17 00:10:52 (GMT +000)
  v2/ansible/modules/core: (detached HEAD cb69744bce) last updated 2014/12/17 00:10:53 (GMT +000)
  v2/ansible/modules/extras: (detached HEAD 8a4f07eecd) last updated 2014/12/17 00:10:54 (GMT +000)
  configured module search path = None
```
##### Environment:

Ubuntu 14.04
##### Summary:

https://github.com/ansible/ansible/commit/7ab1e525755b155cc148abf5b5bba4cb34afb99a just broke this.  Any idea why?

```
Traceback (most recent call last):
  File "/mnt/ansible/bin/ansible-playbook", line 332, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/mnt/ansible/bin/ansible-playbook", line 272, in main
    pb.run()
  File "/mnt/ansible/lib/ansible/playbook/__init__.py", line 313, in run
    play = Play(self, play_ds, play_basedir, vault_password=self.vault_password)
  File "/mnt/ansible/lib/ansible/playbook/play.py", line 120, in __init__
    ds = template(basedir, ds, temp_vars)
  File "/mnt/ansible/lib/ansible/utils/template.py", line 132, in template
    d[k] = template(basedir, v, templatevars, lookup_fatal, depth, expand_lists, convert_bare, fail_on_undefined, filter_fatal)
  File "/mnt/ansible/lib/ansible/utils/template.py", line 132, in template
    d[k] = template(basedir, v, templatevars, lookup_fatal, depth, expand_lists, convert_bare, fail_on_undefined, filter_fatal)
  File "/mnt/ansible/lib/ansible/utils/template.py", line 118, in template
    varname = template_from_string(basedir, varname, templatevars, fail_on_undefined)
  File "/mnt/ansible/lib/ansible/utils/template.py", line 328, in template_from_string
    environment.filters.update(_get_filters())
  File "/mnt/ansible/lib/ansible/utils/template.py", line 51, in _get_filters
    plugins = [ x for x in utils.plugins.filter_loader.all()]
  File "/mnt/ansible/lib/ansible/utils/plugins.py", line 219, in all
    self._module_cache[path] = imp.load_source('.'.join([self.package, name]), path)
  File "/mnt/ansible/lib/ansible/runner/filter_plugins/math.py", line 28, in <module>
    def logarithm(x, base=math.e):
AttributeError: 'module' object has no attribute 'e'
```
##### Steps To Reproduce:

I'm using Ansible on EC2 to bootstrap each instance with a user-data script.  Here's a small snippet that doesn't work.

~/dummy.yml

```

---
- hosts: tag_Name_FooInstance
  remote_user: ubuntu

  tasks:
    - command: hostname
```

```
#!/bin/bash -ex
cd /mnt/ansible
source ./hacking/env-setup
ansible-playbook -i /mnt/ansible/plugins/inventory/ec2.py -c local --limit $(curl http://169.254.169.254/latest/meta-data/public-ipv4) ~/dummy.yml
exit 0
```
##### Expected Results:

Execute hostname.
##### Actual Results:

Error message above.

**Comment by abhinav3295** [comment]
can you show the output of

```
#!/usr/bin/env python
import math
print(math.e)
```

it should be

```
2.71828182846
```

**Comment by abadger** [comment]
Yes, it should have been fixed about 30 minutes later with: c808c8a22ad40de15e1f3877212358fd2eacceb9

The problem was that a new python module was created called math.py  to hold jinja2 filters related to doing math transformations.  The new module used the python stdlib math module to do this.  Unfortunately, by default in python2, relative imports take precedent over absolute imports so the new module attempted to import itself in order to find euler's contant (e).   Since this was in the python stdlib math module, not the new module, this failed.  I remedied it by using
`from__future__ import absolute_import` so that the python stdlib math module would be found.

Thanks for noticing and reporting the breakage so quickly!

**Comment by semifocused** [comment]
> The problem was that a new python module was created called math.py to
> hold jinja2 filters related to doing math transformations.

Another issue that this is presenting for us is that the presence of math.py (in Ansible) precludes imports from the stdlib math module.  So our imports of, e.g. math.floor no longer work, because we're getting the Ansible module instead.  Why does the ansible module need to be called math.py, wouldn't it be better as "ansible_math.py" or somesuch?
